app/web/src/features/dashboard/CommunityBrowser.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Community } from "proto/communities_pb";
+import { service } from "service";
+import hookWrapper from "test/hookWrapper";
+
+import CommunityBrowser from "./CommunityBrowser";
+
+const getCommunityMock = service.communities
+  .getCommunity as jest.MockedFunction<typeof service.communities.getCommunity>;
+const listCommunitiesMock = service.communities
+  .listCommunities as jest.MockedFunction<
+  typeof service.communities.listCommunities
+>;
+
+describe("Community browser", () => {
+  beforeEach(() => {
+    getCommunityMock.mockResolvedValue({
+      communityId: 1,
+      name: "Global",
+      slug: "global",
+    } as Community.AsObject);
+
+    listCommunitiesMock.mockImplementation(async (id) => ({
+      communitiesList: [
+        {
+          communityId: Number.parseInt(`${id}1`),
+          name: `${id}1`,
+          slug: `${id}1`,
+        } as Community.AsObject,
+        {
+          communityId: Number.parseInt(`${id}2`),
+          name: `${id}2`,
+          slug: `${id}2`,
+        } as Community.AsObject,
+      ],
+      nextPageToken: "",
+    }));
+  });
+
+  it("has correct display logic", async () => {
+    render(<CommunityBrowser />, { wrapper: hookWrapper });
+
+    //initial display
+    expect(await screen.findByRole("button", { name: "11" })).toBeVisible();
+    expect(screen.getByRole("link", { name: "Global" })).toBeVisible();
+
+    //clicking down the tree
+    userEvent.click(screen.getByRole("button", { name: "11" }));
```

Comment:
I wonder if it's worth storing the button found from `findByRole` and re-use it here (and repeat this comment for other instances below), since we mentioned before `*ByRole` queries can be quite slow? Seeing there are some thorough checks here, repeating the slow query might add up pretty quickly, especially as it doesn't need to be repeated (except when you're checking for the absence of it in the DOM).

But maybe it's not a big deal if CI is happy with it? 🤷  Would be interesting to see if it makes a big difference to the running time of the test.